File: 2nd year/Laboratories/MAP/L9/CRUD/UI/Tui/MainTui.cs
using System;
using System.Collections.Generic;
using CRUD.Service;

namespace CRUD.UI.Tui
{
    public class MainTui : BaseTui
    {
        public MainTui(
            TaskService taskService,
            StudentService studentService,
            MarkService markService,
            ReportsService reportsService,
            FiltersService filtersService)
        {
            var taskTui = new TaskTui(taskService);
            var studentTui = new StudentTui(studentService);
            var markTui = new MarkTui(markService);
            var reportsTui = new ReportsTui(reportsService);
            var filtersTui = new FiltersTui(filtersService);
            Initialize("Main TUI", new Dictionary<string, Action>
            {
                {"Tasks", taskTui.Run},
                {"Students", studentTui.Run},
                {"Marks", markTui.Run},
                {"Reports", reportsTui.Run},
                {"Filters", filtersTui.Run}
            });
        }
    }
}

File: 2nd year/Laboratories/MAP/7/src/main/java/ui/tui/MainTUI.java
package ui.tui;

import service.MarkService;
import service.StudentService;
import service.TaskService;

import java.util.HashMap;

public class MainTUI extends BaseTUI {
    private MarkTUI markTUI;
    private TaskTUI taskTUI;
    private StudentTUI studentTUI;

    public MainTUI(MarkService markService, TaskService taskService, StudentService studentService) {
        this.markTUI = new MarkTUI(markService);
        this.taskTUI = new TaskTUI(taskService);
        this.studentTUI = new StudentTUI(studentService);
        generateTUI(
                "Main TUI",
                new HashMap<String, Runnable>() {{
                    put("Mark menu", MainTUI.this.markTUI::run);
                    put("Task menu", MainTUI.this.taskTUI::run);
                    put("Student menu", MainTUI.this.studentTUI::run);
                }});
    }
}
